lib/ecto/changeset.ex

```diff
@@ -419,8 +419,14 @@ defmodule Ecto.Changeset do
     end
   end
 
-  defp cast_key(key) when is_binary(key),
-    do: {String.to_existing_atom(key), key}
+  defp cast_key(key) when is_binary(key) do
+    try do
+      {String.to_existing_atom(key), key}
+    rescue
+      ArgumentError ->
+        raise "could not convert the parameter #{key} into an atom, #{key} is not in the schema."
```

Comment:
Let's go with this message please:

    raise ArgumentError, "could not convert the parameter `#{key}` into an atom, `#{key}` is not a schema field"

The changes were:

    1. Wrap keys in backticks
    2. Remove trailing dot
    3. Raise it still as ArgumentError

Can you please amend? Thank you!